internal/controller/ipam/clusteripam_controller.go

```diff
@@ -58,9 +60,19 @@ func (r *ClusterIPAMReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	l.Info("Processing provider specific data")
 	adapterData, err := r.processProvider(ctx, clusterIPAMClaim)
 	if err != nil {
+		apimeta.SetStatusCondition(&clusterIPAMClaim.Status.Conditions,
+			metav1.Condition{
+				Type:               kcmv1.IPAMProviderConditionType,
+				Status:             metav1.ConditionUnknown,
+				Reason:             kcmv1.FailedReason,
+				Message:            fmt.Sprintf("ClusterIPAMClaim provider processing failed: %v", err),
+				ObservedGeneration: clusterIPAMClaim.Generation,
+			})
+
 		return ctrl.Result{}, fmt.Errorf("failed to create provider specific data for ClusterIPAM %s/%s: %w", clusterIPAM.Namespace, clusterIPAM.Name, err)
 	}
 
+	apimeta.RemoveStatusCondition(&clusterIPAMClaim.Status.Conditions, kcmv1.IPAMProviderConditionType)
```

Comment:
do i understand it correctly that the condition behaves like `error or nothing`? if so, then i reckon the name should be changed to something similar to `IPAMProviderConditionError` 

otherwise, the condition must be set (if changed) in case of no prior error